server/game/gameSystems/LookMoverDeckCardsTopOrBottomSystem.ts

```diff
@@ -0,0 +1,81 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardTargetSystem, type ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+import { GameEvent } from '../core/event/GameEvent';
+import { Location } from '../core/Constants';
+import { lookAt, moveCard } from './GameSystemLibrary';
+
+export interface ILookMoveDeckCardsTopOrBottomProperties extends ICardTargetSystemProperties {
+    amount: number;
+}
+
+export class LookMoveDeckCardsTopOrBottomSystem<TContext extends AbilityContext = AbilityContext> extends CardTargetSystem<TContext, ILookMoveDeckCardsTopOrBottomProperties> {
+    public override readonly name = 'LookMoveDeckCardsTopOrBottomSystem';
+
+    protected override defaultProperties: ILookMoveDeckCardsTopOrBottomProperties = {
+        amount: 1,
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public override eventHandler(event): void { }
+
+    public override queueGenerateEventGameSteps(events: GameEvent[], context: TContext): void {
+        const player = context.player;
+        const { amount } = this.generatePropertiesFromContext(context);
+        const deckLength = player.drawDeck.length;
+        if (deckLength === 1) {
+            events.push(lookAt({
+                target: player.drawDeck[0],
+                sendChatMessage: true,
+            }).generateEvent(context.target, context));
+        } else {
+            const actual_amount = amount > deckLength ? deckLength : amount;
+            let cards = player.drawDeck.slice(0, actual_amount);
+
+            // @ts-ignore
+            const choiceHandler = (player) => {
+                if (cards.length > 0) {
+                    context.game.promptWithHandlerMenu(player, {
+                        activePromptTitle: 'Select card to move to the top or bottom of the deck',
+                        choices: cards.map((card: Card) => [
+                            'Put ' + card.internalName + ' on top',
```

Comment:
Internal name is mostly for logging and testing and shouldn't be used in anything that the player will see. Use title instead.

Reply:
changed InternalName into title